test/e2e/tests/tokens/custom-token-add-approve.spec.js

```diff
@@ -135,47 +133,31 @@ describe('Create token, approve token and approve token without gas', function (
             /0x095ea7b30000000000000000000000009bc5baf874d2da8d216ae9f137804184ee5afef4/u,
           ),
         );
-        await driver.clickElement({ text: 'Next', tag: 'button' });
-
-        await driver.findElement({
-          text: 'Spending cap request for your ',
-          css: '.box--flex-direction-row',
-        });
 
-        const defaultSpendingCap = await driver.findElement({
+        await driver.clickElement({ text: 'Next', tag: 'button' });
+        // Spending cap modal is opened
+        await driver.waitForSelector({
           text: '7 TST',
           css: '.mm-box > h6',
         });
 
-        assert.equal(
-          await defaultSpendingCap.getText(),
-          '7 TST',
-          'Default value is not correctly set',
-        );
-
         await driver.clickElement({
           text: 'Approve',
           tag: 'button',
         });
 
+        // We want to wait until txn is finished, and the dialog is closed
+        // before checking it in the expanded view of extension
+        await driver.waitUntilXWindowHandles(2);
+
+        // Moved to expanded window to validate the txn
         await driver.switchToWindow(extension);
         await clickNestedButton(driver, 'Activity');
-
-        // check list of pending transactions in extension
-        await driver.wait(async () => {
-          const pendingTxes = await driver.findElements('.activity-list-item');
-          return pendingTxes.length === 1;
-        }, 10000);
-
-        const approveTokenTask = await driver.waitForSelector({
-          // Selects only the very first transaction list item immediately following the 'Pending' header
-          css: '.transaction-list__completed-transactions .activity-list-item [data-testid="activity-list-item-action"]',
-          text: 'Approve TST spending cap',
-        });
-        assert.equal(
-          await approveTokenTask.getText(),
-          'Approve TST spending cap',
```

Comment:
Same question:  Is this check intended to be deleted?

Reply:
yes, it seems duplicated for me to select by text, then validate the text.. they are duplicated steps...


```

        const approveTokenTask = await driver.waitForSelector({
          // Selects only the very first transaction list item immediately following the 'Pending' header
          css: '.transaction-list__completed-transactions .activity-list-item [data-testid="activity-list-item-action"]',
          text: 'Approve TST spending cap',
        });
        assert.equal(
          await approveTokenTask.getText(),
          'Approve TST spending cap',
```